src/state/apis/nft/nftApi.ts

```diff
@@ -22,31 +27,84 @@ type GetNftCollectionInput = {
 
 const moduleLogger = logger.child({ namespace: ['nftApi'] })
 
+const updateNftItem = (originalItem: NftItem, currentItem: NftItem) => {
+  if (!originalItem.medias.length && currentItem.medias.length) {
+    originalItem.medias = currentItem.medias
+  }
+  if (originalItem.rarityRank === null && typeof currentItem.rarityRank === 'number') {
+    originalItem.rarityRank = currentItem.rarityRank
+  }
+  if (!originalItem.description && currentItem.description) {
+    originalItem.description = currentItem.description
+
+    if (!originalItem.name && currentItem.name) {
+      originalItem.name = currentItem.name
+    }
+  }
+
+  return originalItem
+}
```

Comment:
As we exhaust different services, we may be getting more data - update the original entry with it